avocado/core/requirements/resolver.py

```diff
@@ -0,0 +1,106 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+"""
+Requirements Resolver module.
+"""
+
+from ..nrunner import RUNNERS_REGISTRY_PYTHON_CLASS, Runnable, Task
+
+SUPPORTED_REQUIREMENTS = {}
+
+
+def create_requirements_tasks(tasks):
+    """Entry point for the Requirements Resolver.
+
+    This function isolates the Requirements Resolver object into this module.
+
+    :param tasks: List of tasks
+    :type tasks: list
+    """
+    resolver = RequirementsResolver(tasks)
+    return resolver.create_requirements_tasks()
+
+
+class UnsupportedRequirementType(Exception):
+    pass
+
+
+class RequirementsResolver:
+    """Requirements Resolver class."""
+
+    def __init__(self, tasks):
+        """
+        :param tasks: List of tasks to be resolved
+        :type tasks: list
+        """
+        self._tasks = tasks
+
+    def create_requirements_tasks(self):
+        """Create the list of tasks based on the test requirements."""
+        requirements_tasks = []
+        for task in self._tasks:
+            if not task.runnable.requirements:
+                continue
```

Comment:
Is there an easy way to unify the `task.runnable.requirements` and `task.prereqs` data structures? Of course they are different types and contain different information but the relate to the same thing more or less which could simplify some of the parsing and redundancy of attributes.

Reply:
Yes, I thought about it, but it will require a mass refactor. I'm planning to do that outside of the requirements resolver work, just like the `identifier` renaming I commented on above.

Reply:
Great, so I guess it makes sense as another TODO card then.

Reply:
Sure, as soon as this work is merged, I'll open the issue.